File: app/src/main/java/com/material/androidpro/dashboard/DashboardViewModel.java
package com.material.androidpro.dashboard;

import android.app.Activity;

import androidx.databinding.ObservableField;

public class DashboardViewModel {

    public final ObservableField<DashboardAdapter> mAdapter = new ObservableField<>();

    DashboardViewModel(Activity activity) {
        DashboardAdapter adapter = new DashboardAdapter(DashboardData.list(activity));
        mAdapter.set(adapter);
    }

}

File: app/src/main/java/com/material/androidpro/dashboard/DashboardViewModel.kt
package com.material.androidpro.dashboard

import android.app.Activity
import androidx.databinding.ObservableField

class DashboardViewModel internal constructor(activity: Activity) {

    @JvmField
    val mAdapter = ObservableField<DashboardAdapter>()

    init {
        val adapter = DashboardAdapter(DashboardData.list(activity))
        mAdapter.set(adapter)
    }

}
